### PR TITLE
release 4.0.0

### PR DESCRIPTION
**Pull request by majormoses**
### Description

Release 4.0.0

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by majormoses** [comment]
We did some testing with adopt openjdk and it worked for us. We may need to add something for oracle java to work now that it requires auth to pull from them.

**Comment by damacus** [comment]
Just a bit of cookstlye and you're good to go 👍

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This Pull Request is probably missing tests.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: resources/oracle_install.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
 property :checksum, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
 property :md5, String, regex: /^[0-9a-f]{32}$|^[a-zA-Z0-9]{40,64}$/
 property :app_home, String
-property :app_home_mode, Integer, default: 0755
+property :app_home_mode, Integer, default: 0o755
```
Comment (Xorima) [inline resources/oracle_install.rb]: is this a typo?

Comment (majormoses) [reply resources/oracle_install.rb]: no this is an octal value, this was updated automatically by running rubocop automatically, I can revert if we want.

Comment (tas50) [reply resources/oracle_install.rb]: We should stick with what cookstyle sets here.

Comment (josephholsten) [inline resources/oracle_install.rb]:
```suggestion
property :app_home_mode, Integer, default: 0755
```

Comment (josephholsten) [reply resources/oracle_install.rb]: per https://github.com/chef/cookstyle/blob/ff76052f3e8f6f7a378f8d7e0fbcdcd0cb99ad26/config/upstream.yml#L3640-L3648 (thx @teknofire) either is acceptable.

**Comment by majormoses** [comment]
ready for another approval since pushing the change since pushing commits here dismiss the review.

**Comment by majormoses** [comment]
released: https://supermarket.chef.io/cookbooks/java/versions/4.0.0

**Pull request merged**